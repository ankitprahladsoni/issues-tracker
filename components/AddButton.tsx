import React from 'react';
import { Button, Modal, Text, TouchableHighlight, View } from 'react-native';

export interface IAddButtonProps {
  type: string;
}

export default class AddButton extends React.Component<IAddButtonProps> {
  state = {
    modalVisible: false,
  };

  setModalVisible(visible: boolean) {
    this.setState({ modalVisible: visible });
  }
  render() {
    return (
      <View>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
        >
          <View>
            <Text>Add stuff here</Text>
            <TouchableHighlight
              testID="close"
              onPress={() => this.setModalVisible(!this.state.modalVisible)}
            >
              <Text>Hide Modal</Text>
            </TouchableHighlight>
          </View>
        </Modal>
        <Button
          testID="add-button"
          title="Add"
          onPress={() => this.setModalVisible(true)}
        />
      </View>
    );
  }
}
