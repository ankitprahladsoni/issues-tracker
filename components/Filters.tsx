import React from 'react';
import { Button, View, Modal, Text, TouchableHighlight } from 'react-native';

export interface IFiltersProps {
  type: string;
}

export default class Filters extends React.Component<IFiltersProps> {
  state = {
    modalVisible: false,
  };

  setModalVisible(visible: boolean) {
    this.setState({ modalVisible: visible });
  }

  render() {
    return (
      <View>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
        >
          <View>
            <Text>filters</Text>
            <TouchableHighlight
              onPress={() => this.setModalVisible(!this.state.modalVisible)}
            >
              <Text>Hide Modal</Text>
            </TouchableHighlight>
          </View>
        </Modal>
        <Button
          testID="filters-button"
          title="Filters"
          onPress={() => {
            this.setModalVisible(true);
          }}
        />
      </View>
    );
  }
}
