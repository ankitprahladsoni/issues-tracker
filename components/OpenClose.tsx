import React from 'react';

export interface OpenCloseProps {}

export interface OpenCloseState {}

export default class OpenClose extends React.Component<
  OpenCloseProps,
  OpenCloseState
> {
  constructor(props: OpenCloseProps) {
    super(props);

    this.state = {};
  }

  render() {
    return <div />;
  }
}
