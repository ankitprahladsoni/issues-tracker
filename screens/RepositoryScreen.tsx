import React from 'react';
import { createMaterialTopTabNavigator } from 'react-navigation';
import { MilestoneScreen } from './MilestoneScreen';
import { StoriesScreen } from './StoriesScreen';
import { BugsScreen } from './BugsScreen';

const TabNavigator = createMaterialTopTabNavigator({
  Milestone: MilestoneScreen,
  Stories: StoriesScreen,
  Bugs: BugsScreen,
});

export default class RepositoryScreen extends React.Component {
  static navigationOptions = {
    title: 'Repository!!!',
  };
  render() {
    return <TabNavigator />;
  }
}
