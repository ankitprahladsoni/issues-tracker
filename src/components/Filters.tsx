import React from 'react';
import { Button, Modal, Text, TouchableHighlight, View } from 'react-native';

export interface FiltersProps {
  type: string;
}

export default class Filters extends React.Component<FiltersProps> {
  state = {
    modalVisible: false,
  };

  setModalVisible(visible: boolean) {
    this.setState({ modalVisible: visible });
  }

  render() {
    return (
      <View>
        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.modalVisible}
          onRequestClose={() => {}}
        >
          <View>
            <Text>filters</Text>
            <TouchableHighlight
              testID="close"
              onPress={() => this.setModalVisible(!this.state.modalVisible)}
            >
              <Text>Hide Modal</Text>
            </TouchableHighlight>
          </View>
        </Modal>
        <Button
          testID="filters-button"
          title="Filters"
          onPress={() => this.setModalVisible(true)}
        />
      </View>
    );
  }
}
