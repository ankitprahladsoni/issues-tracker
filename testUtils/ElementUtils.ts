import { ShallowWrapper } from 'enzyme';

export function valueFromWrapper(
  wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>,
  elementId: string
) {
  return wrapper.findWhere(cmd => cmd.props().testID === elementId).props()
    .children;
}

export function propsForElementInsideWrapper(
  wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>,
  elementId: string
) {
  return wrapper.findWhere(cmd => cmd.props().testID === elementId).props();
}
