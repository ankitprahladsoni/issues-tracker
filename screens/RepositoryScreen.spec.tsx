import AddButton from 'components/AddButton';
import Filters from 'components/Filters';
import OpenClose from 'components/OpenClose';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { propsForElementInsideWrapper } from 'testUtils/ElementUtils';
import RepositoryScreen, { TabNavigator } from './RepositoryScreen';

let wrapper: ShallowWrapper;
beforeEach(() => {
  wrapper = shallow(<RepositoryScreen />);
});

describe('For Open/Closed button', () => {
  const wrapper = shallow(<RepositoryScreen />);
  const openClose = wrapper.find(OpenClose).shallow();

  describe.each`
    button      | buttonId           | shouldShowOpen
    ${'Open'}   | ${'open-button'}   | ${true}
    ${'Closed'} | ${'closed-button'} | ${false}
  `('When $button button is pressed', ({ buttonId, shouldShowOpen }) => {
    it('should pass screenProps.shouldShowOpen as $shouldShowOpen', () => {
      propsForElementInsideWrapper(openClose, buttonId).onPress();
      const screenProps = wrapper.find(TabNavigator).props().screenProps;

      expect(screenProps.shouldShowOpen).toBe(shouldShowOpen);
    });
  });
});

it('should have a Filters component', () => {
  expect(wrapper.find(Filters).length).toBe(1);
});

it('should have a AddButton component', () => {
  expect(wrapper.find(AddButton).length).toBe(1);
});
