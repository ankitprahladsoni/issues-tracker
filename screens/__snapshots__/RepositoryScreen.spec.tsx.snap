// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  collapsable={false}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#2196f3",
        "elevation": 4,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 0.5,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 0.5,
        "zIndex": 1,
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffeb3b",
            "bottom": 0,
            "height": 2,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "overflow": "scroll",
        }
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
